pyiron/atomistics/structure/analyse.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+from pyiron_base import Settings
+from sklearn.cluster import AgglomerativeClustering
+
+__author__ = "Joerg Neugebauer, Sam Waseda"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Sam Waseda"
+__email__ = "waseda@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+s = Settings()
+
+class Analyse:
+    """ Class to analyse atom structure.  """
+    def __init__(self, ref_structure):
```

Comment:
Nit: I guess this can just be `structure` unless we wind up creating child structures based off it. 

Also, using `self._structure = structure.copy()` might be safer so no one accidentally modifies a structure in place during analysis.

Reply:
We can't copy the structure, because `structure.analyse.do_something()` should always reflect the current structure.

> Nit: I guess this can just be structure unless we wind up creating child structures based off it.

Is it about the variable name (i.e. are you suggesting to change `ref_structure` to `structure`)?

Reply:
> We can't copy the structure, because `structure.analyse.do_something()` should always reflect the current structure.

You're right, I was not thinking clearly. I mean we always *can* copy a structure, but in this case I now see the foolishness of doing so.

> Is it about the variable name (i.e. are you suggesting to change `ref_structure` to `structure`)?

Yes. I think `ref` here is redundant so let's scratch it.